src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -674,21 +697,27 @@ void AsynchronousMetrics::update(std::chrono::system_clock::time_point update_ti
         {
             Int64 amount = total_memory_tracker.get();
             Int64 peak = total_memory_tracker.getPeak();
-            Int64 new_amount = data.resident;
+            Int64 rss = data.resident;
+            Int64 free_memory_in_allocator_arenas = 0;
+
+#if USE_JEMALLOC
+            /// This is a memory which is kept by allocator.
+            /// Will subsract it from RSS to decrease memory drift.
+            free_memory_in_allocator_arenas = je_malloc_pdirty * getPageSize();
+#endif
 
-            Int64 difference = new_amount - amount;
+            Int64 difference = rss - free_memory_in_allocator_arenas - amount;
 
             /// Log only if difference is high. This is for convenience. The threshold is arbitrary.
             if (difference >= 1048576 || difference <= -1048576)
                 LOG_TRACE(log,
                     "MemoryTracking: was {}, peak {}, will set to {} (RSS), difference: {}",
```

Comment:
Maybe also put `free_memory_in_allocator_arenas` value into the log?

Reply:
No message in the log

Reply:
Yes, sorry.
Well, this value is still available from system table...